sunpy/net/dataretriever/sources/gong_synoptic.py

```diff
@@ -0,0 +1,172 @@
+    2019-12-31 22:00:00 2020-01-01 02:00:00    NSO       GONG        nan
+    <BLANKLINE>
+    <BLANKLINE>
+
+    """
+
+    def _get_url_for_timerange(self, timerange, **kwargs):
+        """
+        Return URL(s) for corresponding timerange.
+
+        Parameters
+        ----------
+        timerange : `~sunpy.time.TimeRange`
+            The time range you want the files for.
+
+        Returns
+        -------
+        `list`
+            The URL(s) for the corresponding timerange.
+        """
+        prefix = 'https://gong2.nso.edu/oQR/zqs/%Y%m/mrzqs%y%m%d/'
+        file_pattern = 'mrzqs\d{6}t\d{4}c\d+_\d+.fits.gz'
+        pref_scrap = Scraper(prefix)
+        directories = pref_scrap.range(timerange)
+        filesurls = []
+        for directory in directories:
+            try:
+                opn = urlopen(directory)
+                try:
+                    soup = BeautifulSoup(opn, "html.parser")
+                    for link in soup.find_all("a"):
+                        href = link.get("href")
+                        if re.match(file_pattern, href):
+                            # extracting timestamp from the filename
+                            timestring = href[5:16]
+                            tstamp = parse_time('20'+timestring.upper()+'00')
+                            if tstamp >= timerange.start and tstamp <= timerange.end:
+                                fullpath = directory + href
+                                filesurls.append(fullpath)
+                finally:
+                    opn.close()
+            except HTTPError as http_err:
+                if http_err.code == 404:
+                    continue
+                raise
+            except Exception:
+                raise
+        return filesurls
+
+    def fetch(self, qres, path=None, error_callback=None, **kwargs):
+        """
+        Download a set of results.
+
+        Parameters
+        ----------
+        qres : `~sunpy.net.dataretriever.QueryResponse`
+            Results to download.
+
+        path : `str`, optional
+            The directory to retrieve the files into.
+
+        Returns
+        -------
+        results: `parfive.Results`
+
+        """
+        if path is not None:
+            path = Path(path)
+
+        urls = [qrblock.url for qrblock in qres]
+
+        filenames = [url.split('/')[-1] for url in urls]
+        local_filenames = [name[:-3] for name in filenames]
+        paths = self._get_full_filenames(qres, filenames, path)
+        local_paths = self._get_full_filenames(qres, local_filenames, path)
+
+        downloader = Downloader()
+
+        for url, filename in zip(urls, paths):
+            downloader.enqueue_file(url, filename=filename)
+
+        resobj = downloader.download()
+
+        for loc_fpath, fpath in zip(local_paths, paths):
+            f = gzip.open(fpath, 'r')
+            g = gzip.open(loc_fpath, 'w')
+            g.write(f.read())
```

Comment:
Why is this needed?

Reply:
Aren't you removing the `.gz` on line 112? why are you then zipping it back without the `gz` suffix?

Reply:
Yes, I removed `.gz` in line 112 to just get what the filename of extracted file should be. I haven't done any extraction there.
`local_fpath` contains path to future extracted file, which is currently empty when I opened it in write mode, and write `f` object to it, which contains the extracted file object in `fits`.

Reply:
sorry I've not commented earlier. I don't understand why you want to unzip the file. I believe fits, and map, reads zipped files fine. I'd prefer to keep the files zipped in my computer.

Reply:
Cool, keeping it zipped then. Thanks for the review!